driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncGridFSBucket.java

```diff
@@ -0,0 +1,376 @@
+        toOutputStream(wrapped.downloadToPublisher(unwrap(clientSession), id), destination);
+    }
+
+    @Override
+    public void downloadToStream(final ClientSession clientSession, final String filename, final OutputStream destination) {
+        downloadToStream(clientSession, filename, destination, new GridFSDownloadOptions());
+    }
+
+    @Override
+    public void downloadToStream(final ClientSession clientSession, final String filename, final OutputStream destination,
+                                 final GridFSDownloadOptions options) {
+        toOutputStream(wrapped.downloadToPublisher(unwrap(clientSession), filename, options), destination);
+    }
+
+    @Override
+    public GridFSFindIterable find() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public GridFSFindIterable find(final Bson filter) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public GridFSFindIterable find(final ClientSession clientSession) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public GridFSFindIterable find(final ClientSession clientSession, final Bson filter) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void delete(final ObjectId id) {
+        delete(new BsonObjectId(id));
+    }
+
+    @Override
+    public void delete(final BsonValue id) {
+        Mono.from(wrapped.delete(id)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void delete(final ClientSession clientSession, final ObjectId id) {
+        delete(clientSession, new BsonObjectId(id));
+    }
+
+    @Override
+    public void delete(final ClientSession clientSession, final BsonValue id) {
+        Mono.from(wrapped.delete(unwrap(clientSession), id)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void rename(final ObjectId id, final String newFilename) {
+        rename(new BsonObjectId(id), newFilename);
+    }
+
+    @Override
+    public void rename(final BsonValue id, final String newFilename) {
+        Mono.from(wrapped.rename(id, newFilename)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void rename(final ClientSession clientSession, final ObjectId id, final String newFilename) {
+        rename(clientSession, new BsonObjectId(id), newFilename);
+    }
+
+    @Override
+    public void rename(final ClientSession clientSession, final BsonValue id, final String newFilename) {
+        Mono.from(wrapped.rename(unwrap(clientSession), id, newFilename)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void drop() {
+        Mono.from(wrapped.drop()).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void drop(final ClientSession clientSession) {
+        Mono.from(wrapped.drop(unwrap(clientSession))).block(TIMEOUT_DURATION);
+    }
+
+    private void toOutputStream(final GridFSDownloadPublisher downloadPublisher, final OutputStream destination) {
```

Comment:
These two methods use Project Reactor Flux to go from synchronous to asynchronous world.